python/pyspark/mllib/linalg.py

```diff
@@ -821,6 +821,23 @@ def __reduce__(self):
             self.numRows, self.numCols, self.values.tostring(),
             int(self.isTransposed))
 
+    def __str__(self):
+        mattoarr = self.toArray()
+        matstr = ""
+        for row in mattoarr:
+            for ind, col in enumerate(row):
+                if ind != 0:
+                    matstr += "  "
+                matstr += str(col)
+            matstr += '\n'
```

Comment:
‘+=’ of string is slow, could you change to '\n'.join(' '.join(row) for row in self.toArray()